cura/CuraPackageManager.py

```diff
@@ -0,0 +1,357 @@
+
+        # Also get all bundled plugins
+        all_metadata = self._plugin_registry.getAllMetaData()
+        for item in all_metadata:
+            plugin_package_info = self.__convertPluginMetadataToPackageMetadata(item)
+            # Only gather the bundled plugins here.
+            package_id = plugin_package_info["package_id"]
+            if package_id in managed_package_id_set:
+                continue
+
+            plugin_package_info["is_bundled"] = True if plugin_package_info["author"]["name"] == "Ultimaker B.V." else False
+            plugin_package_info["is_active"] = self._plugin_registry.isActivePlugin(package_id)
+            package_type = "plugin"
+            if package_type not in installed_packages_dict:
+                installed_packages_dict[package_type] = []
+            installed_packages_dict[package_type].append( plugin_package_info )
+
+        return installed_packages_dict
+
+    def __convertPluginMetadataToPackageMetadata(self, plugin_metadata: dict) -> dict:
+        package_metadata = {"package_id": plugin_metadata["id"],
+                            "package_type": "plugin",
+                            "display_name": plugin_metadata["plugin"]["name"],
+                            "description": plugin_metadata["plugin"].get("description"),
+                            "package_version": plugin_metadata["plugin"]["version"],
+                            "cura_version": int(plugin_metadata["plugin"]["api"]),
+                            "website": "",
+                            "author": {
+                                "name": plugin_metadata["plugin"].get("author", ""),
+                                "email": "",
+                                "website": "",
+                            },
+                            "tags": ["plugin"],
+                            }
+        return package_metadata
+
+    # Checks if the given package is installed.
+    def isPackageInstalled(self, package_id: str) -> bool:
+        return self.getInstalledPackageInfo(package_id) is not None
+
+    # Schedules the given package file to be installed upon the next start.
+    @pyqtSlot(str)
+    def installPackage(self, filename: str) -> None:
+        # Get package information
+        package_info = self.getPackageInfo(filename)
+        package_id = package_info["package_id"]
+
+        has_changes = False
+        # Check the delayed installation and removal lists first
+        if package_id in self._to_remove_package_set:
+            self._to_remove_package_set.remove(package_id)
+            has_changes = True
+
+        # Check if it is installed
+        installed_package_info = self.getInstalledPackageInfo(package_info["package_id"])
+        to_install_package = installed_package_info is None  # Install if the package has not been installed
+        if installed_package_info is not None:
+            # Compare versions and only schedule the installation if the given package is newer
```

Comment:
I think this check should be performed in the GUI or the API, not in the back-end. If we want to support downgrading plug-ins in the future this would prevent us from doing that.